### ISSUE TITLE
Restrict Rating

### ISSUE THREAD

**Issue by SonjaKoenig** [learnweb/moodle-mod_ratingallocate#28]
Currently it is possible to set a value for the maximum of allowed denials of choices. In some applications it may be useful to be able to set a value for the maximum of allowed "highest ratings".
This would mean we need another textfield in the options section for the strategy.

For a start this makes especially sense for the Lickert-Skala.

**Comment by tobiasreischmann** [comment]
I can not think of any case, where this may be necessary. The restriction on maximum allowed rating with 0 is done, because the algorithm does never allocate a student to a choice that student has rated with 0. Thus allowing a student to rate to many choices with 0 results in a high number of unassigned students and a high effort for manuall allocation.

**Comment by SonjaKoenig** [comment]
Just some further explanation with the example of the Lickert-Skala:

Lets say we have 6 possible rankings (0-5). And 20 choices.
We say that the student can only rate 15 choices with 0, so the student HAS to decide on 5.
We now want to restrict the student further, so he can't circumverent our algorithm: we don't want him to rate all of the rest 5.. Therefore we say he can rate only e.g. 2 choices with the highest rank of 5. 
So NOW the student really has to think about what choices he wants and not be like 'hell, I don't care'. This makes sense, because some students DO think about their choices and might loose the fight for their most preffered choice, because someone else thought he doesn't care and wants all of them with highest preference.

Basically it was just a proposal. But I can see that this might be too much effort with too little result.
We can certainly keep this closed.

**Comment by tobiasreischmann** [comment]
If you really want each student to decide, which of the choices he likes most, then you should consider the rank strategy.
Besides, the algorithm tries to optimize the overall satisfaction. Thus, it does not matter if a student rates all choices with 1 or all choices with 5. Because the algorithm will realize, that it doesn't matter for the student, which of the choices he wants.
However, there might be a problem, when the total number spaces are sparse. When one of the two students will not get a space in one of his courses at all, I think the one rating all with 5 will have the advantage.

**Comment by SonjaKoenig** [comment]
That *total number spaces are sparse* is the crux of the matter. Until then it's fair, even if some students don't care, rate all with 5, and get the advantage.
This is just a problem that all allocations have :|
